Automattic/jetpack
# WPCOM no longer forces home_urls to be http

Fixes https://github.com/Automattic/wp-calypso/issues/57308

#### Changes proposed in this Pull Request:
* WPCOM: No longer filter `home_url` to always be http
* Jetpack: No changes.

#### Autogenerated Sync Patch
D70116-code

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Have two simple sites, one with a custom domain, one without
* Visit https://developer.wordpress.com/docs/api/console/
* Use WP.COM API / v1.2 / GET / `/sites/1234`, where 1234 is the site id
* Notice the `url` key for the custom site uses http, but the non-custom one uses https.
* Apply WPCOM patch of this PR: D70116-code
* The custom site should now have an https url.
* You should be able to visit the https version of your site (hosted on wp.com) without seeing errors.

#### Background
This was once removed in 2015 and re-added. The timeline looks to be:
* https://github.com/Automattic/jetpack/pull/2603
  * 2015-08-24: This removes the filter, which is what I want to do, but it had to be re-added.
* p1443108993000912-slack-C02DQP0FP (includes a video)
  * 2015-09-24: "The View Site link in Calypso is forcing HTTPS on custom domains which is resulting in warning messages"
  * The video shows visiting a custom simple site on https but the browser shows a certificate warning.
    * This no longer seems to be the case in my testing.
* r124548-wpcom
  * 2015-09-24: Revert pushed.
    * Reverts r124538-wpcom, which is an automatic sync from jetpack -> wpcom, which includes https://github.com/Automattic/jetpack/commit/51ef6c8
    * This adds the filter back
* https://github.com/Automattic/jetpack/commit/15a7d38e65601c04c973d8afc7e4dd1ea2f36b9d | Belongs to https://github.com/Automattic/jetpack/pull/2803
  * 2015-10-04: The filter was re-added.
  * There's hardly any documentation here :(

The question here is: Do we know that all simple sites hosted on WPCOM with custom domains no longer have certificate warnings? This is definitely something that could have been fixed sometime between 2015 and now, but I don't know what was originally causing the errors, and what fixed it, and if the fix applies to all circumstances.

Answer: In 2016, we offered "free HTTPS for all custom domains hosted on WordPress.com". Blog post: https://wordpress.com/blog/2016/04/08/https-everywhere-encryption-for-all-wordpress-com-sites/
